rust/connlib/libs/common/src/session.rs

```diff
@@ -251,6 +253,16 @@ where
             }));
         }
 
+        tracing::info!("Starting connlib!");
+        tracing::info!("Saving log files to: {log_dir}");
+        tracing::info!("Debug mode: {debug_mode}");
+
+        // TODO: Extract this to a logging module
+        // TODO: Check if logDir is sane (e.g. not an empty string)
+        let file_appender = tracing_appender::rolling::hourly(log_dir, "connlib.log");
+        let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
+        tracing_subscriber::fmt().with_writer(non_blocking).init();
```

Comment:
`tracing_subscriber` should be configured in the binaries not in the libraries, this will grant us more flexibility and reduce a bug if the binaries configure it thsemselves.

Furthermore, I think this will consume the logs and prevent us from seeing anything in stdout so we need to do something like I did [here](https://github.com/firezone/firezone-gateway/blob/main/gateway/src/logging.rs#L37-L63) with the old gateway.

Reply:
Hmm, gotcha. I've updated the PR to move all init logging logic into the clients' entrypoints (`main.rs` or `lib.rs`). From there we can set up `tracing-subscriber` appropriately, and then the rest of the `Session.connect` flow need not concern itself with this stuff.

I still need to split off the logs to STDOUT and the file like you mentioned above.

Reply:
This has been updated to follow the approach you have in the old gateway using a `tracing_subscriber::registry()` with multiple layers to write to file and still log to the OS-appropriate log system.